tests/0081-admin.c

```diff
@@ -3652,239 +3652,335 @@ static void do_test_ListConsumerGroupOffsets(const char *what,
+
+        TEST_CALL_ERR__(rd_kafka_AdminOptions_set_request_timeout(
+            options, 30 * 1000 /* 30s */, errstr, sizeof(errstr)));
+
+        rd_kafka_AlterUserScramCredentials(
+            rk, alterations, RD_ARRAY_SIZE(alterations), options, queue);
         rd_kafka_AdminOptions_destroy(options);
+        rd_kafka_UserScramCredentialAlteration_destroy_array(
+            alterations, RD_ARRAY_SIZE(alterations));
+
         /* Wait for results */
         event = rd_kafka_queue_poll(queue, -1 /*indefinitely*/);
-        if (rd_kafka_event_error(event)) {
-                /* Request failed */
-                TEST_FAIL("Event Errored : AlterUserScramCredentials[ScramConfigAdmin]");
-                fprintf(stderr, "%% AlterUserScramCredentials failed: %s\n",
-                        rd_kafka_event_error_string(event));
-        } else {
-                const rd_kafka_AlterUserScramCredentials_result_t *result;
-                result  = rd_kafka_event_AlterUserScramCredentials_result(event);
-                size_t num_results = rd_kafka_AlterUserScramCredentials_result_get_count(result); 
-                size_t i;
-                /* num_results should be 1*/
-                TEST_ASSERT(num_results==1,"There should be only 1 Result for a User !");
-                for (i = 0; i < num_results; i++){
-                        rd_kafka_UserScramCredentialAlterationResultElement_t *element = rd_kafka_AlterUserScramCredentials_result_get_element(result,i);
-                        char *username;
-                        rd_kafka_error_t *error;
-                        username = rd_kafka_UserScramCredentialAlterationResultElement_get_user(element);
-                        error = rd_kafka_UserScramCredentialAlterationResultElement_get_error(element);
-                        rd_kafka_resp_err_t errorcode = rd_kafka_error_code(error);
-                        /* username should be broker and errorcode should be 0*/
-                        TEST_ASSERT(strcmp(users[0],username)==0,"Username does not match !");
-                        TEST_ASSERT(errorcode==0,"Error Code should be 0");
-
-                }
+        err   = rd_kafka_event_error(event);
+#if !WITH_SSL
+        TEST_ASSERT(err == RD_KAFKA_RESP_ERR__INVALID_ARG,
+                    "Expected _INVALID_ARG, not %s", rd_kafka_err2name(err));
+        rd_kafka_event_destroy(event);
+        goto final_checks;
+#else
+        TEST_ASSERT(err == RD_KAFKA_RESP_ERR_NO_ERROR,
+                    "Expected NO_ERROR, not %s", rd_kafka_err2name(err));
+
+        alter_result = rd_kafka_event_AlterUserScramCredentials_result(event);
+        alter_responses = rd_kafka_AlterUserScramCredentials_result_responses(
+            alter_result, &response_cnt);
+        /* response_cnt should be 1*/
+        TEST_ASSERT(response_cnt == 1,
+                    "There should be exactly 1 response, got %" PRIusz,
+                    response_cnt);
+        for (i = 0; i < response_cnt; i++) {
```

Comment:
nit: For loop is redundant if we're asserting `response_cnt == 1`, it's clearer to just use [0] and proceed with that.

Reply:
Applicable here and at other places